main/guides/getting-started/swapaoo-making-a-payment-explainer.md

```diff
@@ -0,0 +1,63 @@
+# Making Payments in Agoric Smart Contracts
+In this document, we'll explain how to send a payment to someone using their `agoric1...` address from an Agoric smart contract using a deposit facet.
+
+## Using a depositFacet
+Let's take a look at the following code snippet from the Swaparoo contract from [dapp-agoric-basics](https://github.com/Agoric/dapp-agoric-basics):
+```js
+const secondDepositFacet = await E(depositFacetFromAddr).lookup(
+  secondPartyAddress,
+  'depositFacet',
+);
+
+await E(secondDepositFacet).receive(secondSeatInvitation);
+
+return 'invitation sent';
+```
+
+## Step-by-Step Explanation
+### Retrieving the Deposit Facet:
+- `depositFacetFromAddr` is an object that provides a lookup function for deposit facets associated with addresses. The Swaparoo contract is provided with a `depositFacetFromAddr` by the proposal (`swaparoo.proposal.js`) as a parameter in the `startGovernedIntance` call (see below):
+  - ```js
+    const it = await startMyGovernedInstance(
+      {
+        zoe,
+        governedContractInstallation: installationP,
+        label: contractName,
+        terms: {},
+        privateArgs: {
+          storageNode,
+          marshaller,
+          namesByAddressAdmin,
```

Comment:
`depositFacetFromAddr` isn't provided directly by `startGovernedIntance`. What's provided directly is `namesByAddressAdmin`.

What it _should_ get is `namesByAddress`, the read-only facet. But there's a bug: https://github.com/Agoric/agoric-sdk/issues/8408 . To work around it, we pass the admin facet and use `fixHub(...)` to make a `namesByAddress` work-alike.

Unfortunately, we give it a different name, `depositFacetFromAddr`. Here's hoping, again, that I find a moment to change that.

The need for this work-around is disappearing soon - https://github.com/Agoric/agoric-sdk/issues/8408 is fixed in [upgrade-14](https://community.agoric.com/t/mainnet-upgrade-agoric-upgrade-14-interchain-stack-smart-wallet-upgrades/633/10), which should hit mainnet around Apr 2.